test/src/test/java/hudson/model/ProjectTest.java

```diff
@@ -175,9 +175,9 @@ public void testGetAssignedLabel() throws Exception{
     public void testGetAssignedLabelString() throws Exception{
         FreeStyleProject p = j.createFreeStyleProject("project");
         Slave slave = j.createOnlineSlave();
-        assertNull("Project should not have any label.", p.getAssignedLabelString());
+        assertEquals("Project should have jenkins master label.", j.jenkins.getSelfLabel().name, p.getAssignedLabelString());
         p.setAssignedLabel(j.jenkins.getSelfLabel());
-        assertNull("Project should return null, because assigned label is Jenkins.", p.getAssignedLabelString());
+        assertEquals("Project should have jenkins master label after set explicitly", j.jenkins.getSelfLabel().name, p.getAssignedLabelString());
```

Comment:
there is a missing test case here - if the project is saved again it should not have any assigned label.